### PR TITLE
`ImageIcon/Source`, basic `AnimatedIcon/Source` support for fallback

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

- `ImageIcon/Source` not supported
- `AnimatedIcon/Source` not supported

## What is the new behavior?

- `ImageIcon/Source` supported
- `AnimatedIcon/Source` code included, but animations not supported at the moment. Instead, Uno always uses `FallbackIconSource`

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/BitmapIconSource.cs [hunk 1]
```diff
@@ -27,9 +27,9 @@ public bool ShowAsMonochrome
 		}
 
 		public static DependencyProperty ShowAsMonochromeProperty { get; } =
-			DependencyProperty.Register(nameof(ShowAsMonochrome), typeof(bool), typeof(BitmapIconSource), new PropertyMetadata(default(bool)));
+			DependencyProperty.Register(nameof(ShowAsMonochrome), typeof(bool), typeof(BitmapIconSource), new PropertyMetadata(default(bool), OnPropertyChanged));
 
-		public override IconElement CreateIconElement()
+		internal protected override IconElement CreateIconElementCore()
```
Comment (davidjohnoliver) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/BitmapIconSource.cs:32]: should this be `internal protected` (ie public) or `private protected` (private)?

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/BitmapIconSource.cs [hunk 2]
```diff
@@ -50,5 +50,19 @@ public override IconElement CreateIconElement()
 
 			return bitmapIcon;
 		}
+
+		internal protected override DependencyProperty GetIconElementPropertyCore(DependencyProperty sourceProperty)
```
Comment (davidjohnoliver) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/BitmapIconSource.cs:54]: ditto

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/FontIconSource.cs [hunk 3]
```diff
@@ -73,10 +73,10 @@ public FontFamily FontFamily
 		}
 
 		public static DependencyProperty FontFamilyProperty { get; } =
-			DependencyProperty.Register(nameof(FontFamily), typeof(FontFamily), typeof(FontIconSource), new PropertyMetadata(new FontFamily(Uno.UI.FeatureConfiguration.Font.SymbolsFont)));
+			DependencyProperty.Register(nameof(FontFamily), typeof(FontFamily), typeof(FontIconSource), new PropertyMetadata(new FontFamily(Uno.UI.FeatureConfiguration.Font.SymbolsFont), OnPropertyChanged));
 
 		/// <inheritdoc />
-		public override IconElement CreateIconElement()
+		internal protected override IconElement CreateIconElementCore()
```
Comment (davidjohnoliver) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/FontIconSource.cs:79]: ditto

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/FontIconSource.cs [hunk 4]
```diff
@@ -100,5 +100,39 @@ public override IconElement CreateIconElement()
 
 			return fontIcon;
 		}
+
+		internal protected override DependencyProperty GetIconElementPropertyCore(DependencyProperty sourceProperty)
```
Comment (davidjohnoliver) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/IconSource/FontIconSource.cs:104]: ditto

**Review by davidjohnoliver — APPROVED** [review]

**Review by carldebilly — APPROVED** [review]

**Pull request merged**